File: runinator-command-center/src/core/domain/models/auth/grant.ts
import type { PermissionLevel, PrincipalType } from "./permission";

export interface Grant {
  id: string | null;
  resource_type: string;
  resource_id: string;
  principal_type: PrincipalType;
  principal_id: string;
  permission: PermissionLevel;
  created_at: string;
}

File: runinator-models/src/auth/grant.rs
#[allow(unused_imports)]
use super::*;

/// a single access grant on a resource.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Grant {
    pub id: Option<Uuid>,
    pub resource_type: ResourceType,
    pub resource_id: Uuid,
    pub principal_type: PrincipalType,
    pub principal_id: Uuid,
    pub permission: Permission,
    pub created_at: DateTime<Utc>,
}
